### ISSUE TITLE
Mendix 5.16 compatibility

### ISSUE THREAD

**Issue by Markos1** [mendix/RestServices#60]
Hi Michel,

If I install the RestServices module in Mendix version 5.16 I cannot run my project anymore ('compilation of Java actions failed'). I think this has to do with the file I have to overrun when adding this module (userlib\commons-httpclient-3.1.jar). Could you help me make the Restservices compatible in Mendix 5.16? I downgraded to Mendix version 5.12 but in that case I cannot run my project anymore  on which I need to build the services so I need to include the RestServices module in Version 5.16. Thanks

Mark

**Comment by mweststrate** [comment]
Can you elaborate on the errors you get? Stacktraces and such?

**Comment by Markos1** [comment]
The error and the stacktrace added.

Mark
________________________________
Van: Michel Weststrate [notifications@github.com]
Verzonden: vrijdag 12 juni 2015 17:40
Aan: mendix/RestServices
CC: Peeters, M.J.P.G.
Onderwerp: Re: [RestServices] Mendix 5.16 compatibility (#60)

Can you elaborate on the errors you get? Stacktraces and such?

On Fri, Jun 12, 2015 at 4:12 PM, Markos1 <notifications@github.com> wrote:

> Hi Michel,
>
> If I install the RestServices module in Mendix version 5.16 I cannot run
> my project anymore ('compilation of Java actions failed'). I think this has
> to do with the file I have to overrun when adding this module
> (userlib\commons-httpclient-3.1.jar). Could you help me make the
> Restservices compatible in Mendix 5.16? I downgraded to Mendix version 5.12
> but in that case I cannot run my project anymore on which I need to build
> the services so I need to include the RestServices module in Version 5.16.
> Thanks
>
> Mark
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mendix/RestServices/issues/60>.
>

—
Reply to this email directly or view it on GitHub<https://github.com/mendix/RestServices/issues/60#issuecomment-111531172>.

...
Buildfile: ...\Feedback system_final 12-06\deployment\build_core.xml

compile:
    [javac] Compiling 91 source files to ...\Feedback system_final 12-06\deployment\run\bin
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\RestServiceHandler.java:7: error: package org.apache.commons.lang does not exist
    [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
    [javac]                                      ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\RestServiceHandler.java:7: error: static import only from classes and interfaces
    [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
    [javac] ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\RestServiceHandler.java:19: error: package org.apache.commons.lang.builder does not exist
    [javac] import org.apache.commons.lang.builder.ReflectionToStringBuilder;
    [javac]                                       ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\RestServiceHandler.java:20: error: package org.apache.commons.lang.builder does not exist
    [javac] import org.apache.commons.lang.builder.ToStringStyle;
    [javac]                                       ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\RestServiceHandler.java:21: error: package org.apache.commons.lang.exception does not exist
    [javac] import org.apache.commons.lang.exception.ExceptionUtils;
    [javac]                                         ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\ChangeLogManager.java:10: error: package org.apache.commons.lang does not exist
    [javac] import org.apache.commons.lang.StringUtils;
    [javac]                               ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\util\DataWriter.java:6: error: package org.apache.commons.lang does not exist
    [javac] import org.apache.commons.lang.StringEscapeUtils;
    [javac]                               ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\consume\ChangeLogListener.java:5: error: package org.apache.commons.lang does not exist
    [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
    [javac]                                      ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\consume\ChangeLogListener.java:5: error: static import only from classes and interfaces
    [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
    [javac] ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\ConsistencyChecker.java:7: error: package org.apache.commons.lang does not exist
    [javac] import org.apache.commons.lang.StringUtils;
    [javac]                               ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\ConsistencyChecker.java:8: error: package org.apache.commons.lang.exception does not exist
    [javac] import org.apache.commons.lang.exception.ExceptionUtils;
    [javac]                                         ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\MicroflowService.java:21: error: package org.apache.commons.lang does not exist
    [javac] import org.apache.commons.lang.StringUtils;
    [javac]                               ^
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\RestServiceHandler.java:68: error: cannot find symbol
    [javac] 			return ReflectionToStringBuilder.toString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    [javac] 			                                                ^
    [javac]   symbol:   variable ToStringStyle
    [javac]   location: class HandlerRegistration
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\RestServiceHandler.java:68: error: cannot find symbol
    [javac] 			return ReflectionToStringBuilder.toString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    [javac] 			       ^
    [javac]   symbol:   variable ReflectionToStringBuilder
    [javac]   location: class HandlerRegistration
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\RestServiceHandler.java:291: error: cannot find symbol
    [javac] 			Throwable cause = ExceptionUtils.getRootCause(e);
    [javac] 			                  ^
    [javac]   symbol:   variable ExceptionUtils
    [javac]   location: class RestServiceHandler
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\RestServiceHandler.java:370: error: cannot find symbol
    [javac] 		checkArgument(isNotEmpty(serviceBaseUrl));
    [javac] 		              ^
    [javac]   symbol:   method isNotEmpty(String)
    [javac]   location: class RestServiceHandler
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\ChangeLogManager.java:455: error: cannot find symbol
    [javac] 		return StringUtils.join(new String[] {
    [javac] 		       ^
    [javac]   symbol:   variable StringUtils
    [javac]   location: class ChangeLogManager
    [javac] ...\Feedback system_final 12-06\javasource\restservices\util\DataWriter.java:112: error: cannot find symbol
    [javac] 			write("\n<tr><td>").write(StringEscapeUtils.escapeHtml(s.key)).write("</td><td>");
    [javac] 			                          ^
    [javac]   symbol:   variable StringEscapeUtils
    [javac]   location: class DataWriter
    [javac] ...\Feedback system_final 12-06\javasource\restservices\util\DataWriter.java:155: error: cannot find symbol
    [javac] 				write(StringEscapeUtils.escapeXml(value));
    [javac] 				      ^
    [javac]   symbol:   variable StringEscapeUtils
    [javac]   location: class DataWriter
    [javac] ...\Feedback system_final 12-06\javasource\restservices\util\DataWriter.java:157: error: cannot find symbol
    [javac] 				write(Utils.autoGenerateLink(StringEscapeUtils.escapeHtml(value)));
    [javac] 				                             ^
    [javac]   symbol:   variable StringEscapeUtils
    [javac]   location: class DataWriter
    [javac] ...\Feedback system_final 12-06\javasource\restservices\consume\ChangeLogListener.java:61: error: cannot find symbol
    [javac] 		checkArgument(isNotEmpty(onUpdateMF), "On update should be non empty");
    [javac] 		              ^
    [javac]   symbol:   method isNotEmpty(String)
    [javac]   location: class ChangeLogListener
    [javac] ...\Feedback system_final 12-06\javasource\restservices\consume\ChangeLogListener.java:62: error: cannot find symbol
    [javac] 		checkArgument(isNotEmpty(onDeleteMF), "On delete should be non empty");
    [javac] 		              ^
    [javac]   symbol:   method isNotEmpty(String)
    [javac]   location: class ChangeLogListener
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\ConsistencyChecker.java:55: error: cannot find symbol
    [javac] 		return errors.size() == 0 ? null : "* " + StringUtils.join(errors, "\n* ");
    [javac] 		                                          ^
    [javac]   symbol:   variable StringUtils
    [javac]   location: class ConsistencyChecker
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\ConsistencyChecker.java:143: error: cannot find symbol
    [javac] 					errors.add("Constraint is not a valid xpath query: " + ExceptionUtils.getRootCauseMessage(e));
    [javac] 					                                                       ^
    [javac]   symbol:   variable ExceptionUtils
    [javac]   location: class ConsistencyChecker
    [javac] ...\Feedback system_final 12-06\javasource\restservices\publish\MicroflowService.java:247: error: cannot find symbol
    [javac] 			data = new JSONObject(StringUtils.isEmpty(body) ? "{}" : body);
    [javac] 			                      ^
    [javac]   symbol:   variable StringUtils
    [javac]   location: class MicroflowService
    [javac] Note: ...\Feedback system_final 12-06\javasource\com\mendix\core\Core.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 25 errors

**Comment by mweststrate** [comment]
It seems that the apache commons-lang module is missing. Can you list which
files there are in the userlib of your project? Re-importing the modules
community commons and rest services from the appstore might solve this.

**Comment by Markos1** [comment]
Sorry for the late reply. I added CommunityCommons and RestServices as modules. Re-importing does not help. I just ran a test with a blank project (Mendix 5.16) and added these two modules resulting in the same error as with the project I am working on when I try to run the project: 'Compilation of Java actions failed'.
________________________________
Van: Michel Weststrate [notifications@github.com]
Verzonden: vrijdag 12 juni 2015 21:08
Aan: mendix/RestServices
CC: Peeters, M.J.P.G.
Onderwerp: Re: [RestServices] Mendix 5.16 compatibility (#60)

It seems that the apache commons-lang module is missing. Can you list which
files there are in the userlib of your project? Re-importing the modules
community commons and rest services from the appstore might solve this.

On Fri, Jun 12, 2015 at 8:40 PM, Markos1 <notifications@github.com> wrote:

> The error and the stacktrace added.
>
> Mark
> ________________________________
> Van: Michel Weststrate [notifications@github.com]
> Verzonden: vrijdag 12 juni 2015 17:40
> Aan: mendix/RestServices
> CC: Peeters, M.J.P.G.
> Onderwerp: Re: [RestServices] Mendix 5.16 compatibility (#60)
>
> Can you elaborate on the errors you get? Stacktraces and such?
>
> On Fri, Jun 12, 2015 at 4:12 PM, Markos1 <notifications@github.com> wrote:
>
> > Hi Michel,
> >
> > If I install the RestServices module in Mendix version 5.16 I cannot run
> > my project anymore ('compilation of Java actions failed'). I think this
> has
> > to do with the file I have to overrun when adding this module
> > (userlib\commons-httpclient-3.1.jar). Could you help me make the
> > Restservices compatible in Mendix 5.16? I downgraded to Mendix version
> 5.12
> > but in that case I cannot run my project anymore on which I need to build
> > the services so I need to include the RestServices module in Version
> 5.16.
> > Thanks
> >
> > Mark
> >
> > �
> > Reply to this email directly or view it on GitHub
> > <https://github.com/mendix/RestServices/issues/60>.
> >
>
> �
> Reply to this email directly or view it on GitHub<
> https://github.com/mendix/RestServices/issues/60#issuecomment-111531172>.
>
> ...
> Buildfile: ...\Feedback system_final 12-06\deployment\build_core.xml
>
> compile:
> [javac] Compiling 91 source files to ...\Feedback system_final
> 12-06\deployment\run\bin
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\RestServiceHandler.java:7: error:
> package org.apache.commons.lang does not exist
> [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\RestServiceHandler.java:7: error:
> static import only from classes and interfaces
> [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\RestServiceHandler.java:19: error:
> package org.apache.commons.lang.builder does not exist
> [javac] import org.apache.commons.lang.builder.ReflectionToStringBuilder;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\RestServiceHandler.java:20: error:
> package org.apache.commons.lang.builder does not exist
> [javac] import org.apache.commons.lang.builder.ToStringStyle;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\RestServiceHandler.java:21: error:
> package org.apache.commons.lang.exception does not exist
> [javac] import org.apache.commons.lang.exception.ExceptionUtils;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\ChangeLogManager.java:10: error:
> package org.apache.commons.lang does not exist
> [javac] import org.apache.commons.lang.StringUtils;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\util\DataWriter.java:6: error: package
> org.apache.commons.lang does not exist
> [javac] import org.apache.commons.lang.StringEscapeUtils;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\consume\ChangeLogListener.java:5: error:
> package org.apache.commons.lang does not exist
> [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\consume\ChangeLogListener.java:5: error:
> static import only from classes and interfaces
> [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\ConsistencyChecker.java:7: error:
> package org.apache.commons.lang does not exist
> [javac] import org.apache.commons.lang.StringUtils;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\ConsistencyChecker.java:8: error:
> package org.apache.commons.lang.exception does not exist
> [javac] import org.apache.commons.lang.exception.ExceptionUtils;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\MicroflowService.java:21: error:
> package org.apache.commons.lang does not exist
> [javac] import org.apache.commons.lang.StringUtils;
> [javac] ^
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\RestServiceHandler.java:68: error:
> cannot find symbol
> [javac] return ReflectionToStringBuilder.toString(this,
> ToStringStyle.SHORT_PREFIX_STYLE);
> [javac] ^
> [javac] symbol: variable ToStringStyle
> [javac] location: class HandlerRegistration
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\RestServiceHandler.java:68: error:
> cannot find symbol
> [javac] return ReflectionToStringBuilder.toString(this,
> ToStringStyle.SHORT_PREFIX_STYLE);
> [javac] ^
> [javac] symbol: variable ReflectionToStringBuilder
> [javac] location: class HandlerRegistration
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\RestServiceHandler.java:291: error:
> cannot find symbol
> [javac] Throwable cause = ExceptionUtils.getRootCause(e);
> [javac] ^
> [javac] symbol: variable ExceptionUtils
> [javac] location: class RestServiceHandler
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\RestServiceHandler.java:370: error:
> cannot find symbol
> [javac] checkArgument(isNotEmpty(serviceBaseUrl));
> [javac] ^
> [javac] symbol: method isNotEmpty(String)
> [javac] location: class RestServiceHandler
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\ChangeLogManager.java:455: error:
> cannot find symbol
> [javac] return StringUtils.join(new String[] {
> [javac] ^
> [javac] symbol: variable StringUtils
> [javac] location: class ChangeLogManager
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\util\DataWriter.java:112: error: cannot find
> symbol
> [javac]
> write("\n<tr><td>").write(StringEscapeUtils.escapeHtml(s.key)).write("</td><td>");
>
> [javac] ^
> [javac] symbol: variable StringEscapeUtils
> [javac] location: class DataWriter
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\util\DataWriter.java:155: error: cannot find
> symbol
> [javac] write(StringEscapeUtils.escapeXml(value));
> [javac] ^
> [javac] symbol: variable StringEscapeUtils
> [javac] location: class DataWriter
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\util\DataWriter.java:157: error: cannot find
> symbol
> [javac]
> write(Utils.autoGenerateLink(StringEscapeUtils.escapeHtml(value)));
> [javac] ^
> [javac] symbol: variable StringEscapeUtils
> [javac] location: class DataWriter
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\consume\ChangeLogListener.java:61: error:
> cannot find symbol
> [javac] checkArgument(isNotEmpty(onUpdateMF), "On update should be non
> empty");
> [javac] ^
> [javac] symbol: method isNotEmpty(String)
> [javac] location: class ChangeLogListener
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\consume\ChangeLogListener.java:62: error:
> cannot find symbol
> [javac] checkArgument(isNotEmpty(onDeleteMF), "On delete should be non
> empty");
> [javac] ^
> [javac] symbol: method isNotEmpty(String)
> [javac] location: class ChangeLogListener
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\ConsistencyChecker.java:55: error:
> cannot find symbol
> [javac] return errors.size() == 0 ? null : "* " + StringUtils.join(errors,
> "\n* ");
> [javac] ^
> [javac] symbol: variable StringUtils
> [javac] location: class ConsistencyChecker
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\ConsistencyChecker.java:143: error:
> cannot find symbol
> [javac] errors.add("Constraint is not a valid xpath query: " +
> ExceptionUtils.getRootCauseMessage(e));
> [javac] ^
> [javac] symbol: variable ExceptionUtils
> [javac] location: class ConsistencyChecker
> [javac] ...\Feedback system_final
> 12-06\javasource\restservices\publish\MicroflowService.java:247: error:
> cannot find symbol
> [javac] data = new JSONObject(StringUtils.isEmpty(body) ? "{}" : body);
> [javac] ^
> [javac] symbol: variable StringUtils
> [javac] location: class MicroflowService
> [javac] Note: ...\Feedback system_final
> 12-06\javasource\com\mendix\core\Core.java uses unchecked or unsafe
> operations.
> [javac] Note: Recompile with -Xlint:unchecked for details.
> [javac] 25 errors
>
> �
> Reply to this email directly or view it on GitHub
> <https://github.com/mendix/RestServices/issues/60#issuecomment-111584557>.
>

�

**Comment by mweststrate** [comment]
I've just tested the module in Mendix 5.16.1, and it is running without any issues. Can you verify the following as listed in https://github.com/mendix/RestServices/tree/master/userlib are present in your userlib? (junit-4.11.jar is not needed)

**Comment by Markos1** [comment]
Finally got it working. I installed some other modules (EmailTemplate for example), re-installed RestServices and all of the sudden I managed to run it the app.

Thanks for your help Michel
________________________________
Van: Michel Weststrate [notifications@github.com]
Verzonden: donderdag 18 juni 2015 7:58
Aan: mendix/RestServices
CC: Peeters, M.J.P.G.
Onderwerp: Re: [RestServices] Mendix 5.16 compatibility (#60)


I've just tested the module in Mendix 5.16.1, and it is running without any issues. Can you verify the following as listed in https://github.com/mendix/RestServices/tree/master/userlib are present in your userlib? (junit-4.11.jar is not needed)

�

**Comment by ghartwich** [comment]
Hi Guys,

I've been experiencing an exact copy of the problem (using 5.16.1), but have not been able to get it working.

**Comment by ghartwich** [comment]
Hi Michel,

I've here a screenshot of my userlib:
![2015625-17059](https://cloud.githubusercontent.com/assets/9248651/8357596/f53993f8-1b5b-11e5-803b-ed5b5bb8cdc7.png)

The beginning of the stack trace:
Buildfile: C:\Users\Laptop\Documents\Mendix Projects\TestApp\deployment\build_core.xml

compile:
    [javac] Compiling 222 source files to C:\Users\Laptop\Documents\Mendix Projects\TestApp\deployment\run\bin
    [javac] C:\Users\Laptop\Documents\Mendix Projects\TestApp\javasource\restservices\publish\RestServiceHandler.java:7: error: package org.apache.commons.lang does not exist
    [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
    [javac]                            

Do you know how I can fix this?

**Comment by RCatersels** [comment]
I think you are missing the commons-lang jar. https://commons.apache.org/proper/commons-lang/
Try adding that one to the userlib.

**Comment by ghartwich** [comment]
Thanks Ronald, but unfortunately no change.

Stack trace the same. I've added screenshot of current userlib below.

Buildfile: C:\Users\Laptop\Documents\Mendix Projects\TestApp\deployment\build_core.xml

compile:
    [javac] Compiling 221 source files to C:\Users\Laptop\Documents\Mendix Projects\TestApp\deployment\run\bin
    [javac] C:\Users\Laptop\Documents\Mendix Projects\TestApp\javasource\restservices\publish\RestServiceHandler.java:7: error: package org.apache.commons.lang does not exist
    [javac] import static org.apache.commons.lang.StringUtils.isNotEmpty;
    [javac]                                      ^

![2015625-172655](https://cloud.githubusercontent.com/assets/9248651/8358235/6927942e-1b5f-11e5-8554-b51e5b70d091.png)

**Comment by ghartwich** [comment]
The problem might have to do when you download the current REST services module (or perhaps the CC module). I created a project (blank!) in the 5.15.1 modeler and downloaded the REST and CC module and it gave me the same compile error.

But if I open 5.16.1 and create a project by downloading the Mendix UX application that comes with REST and CC included there are no compile error.

Is this helpful towards a solution?
Gonzo

**Comment by RoelandSalij** [comment]
Which versions of Community commons all of you use?
Recently libraries in Community commons have been updated to more recent versions.

**Comment by ghartwich** [comment]
Hi Roeland, The problem arises with the latest Community Commons version 5.4. I started a blank project and downloaded the current Community Commons module and the current REST module and I got the compile errors. This happens both in 5.15.1 and 5.16.1.

In comparison I downloaded the Mendix UX application that includes Community Commons 5.2.1 and the REST module and that gives no problem in the 5.16.1 modeler.
